lib/virt_autotest/utils.pm

```diff
@@ -42,62 +42,89 @@ sub trim {
 }
 
+# Usage: check_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
+# to specify daemons which will be checked: check_modular_libvirt_daemons(qemu storage ...)
+# to check all required modular daemons without any daemons passed
+sub check_modular_libvirt_daemons {
+    my @daemons = @_;
+
+    if (!@daemons) {
+        @daemons = qw(network nodedev nwfilter secret storage proxy log lock);
+        (is_xen_host) ? push @daemons, 'xen' : push @daemons, 'qemu';
+    }
+
+    foreach my $daemon (@daemons) {
+        systemctl("status virt${daemon}d.service");
+        if (($daemon == 'lock') || ($daemon == 'log')) {
+            systemctl("status virt${daemon}d\{,-admin\}.socket");
+        } else {
+            systemctl("status virt${daemon}d\{,-ro,-admin\}.socket");
+        }
+    }
+
+    record_info("Modular Libvirt daemons checked", join(' ', @daemons));
+    save_screenshot;
+}
+
 # Usage: restart_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
 # to specify daemons which will be restarted: restart_modular_libvirt_daemons(virtqemud virtstoraged ...)
 # to restart all modular daemons without any daemons passed
 sub restart_modular_libvirt_daemons {
-    my @daemons;
+    my @daemons = @_;
 
-    if (@_ == 0) {
-        if (is_xen_host) {
-            @daemons = qw(virtxend virtstoraged virtnetworkd virtnodedevd virtsecretd virtproxyd virtnwfilterd);
-        } else {
-            @daemons = qw(virtqemud virtstoraged virtnetworkd virtnodedevd virtsecretd virtproxyd virtnwfilterd);
-        }
-    } else {
-        @daemons = @_;
+    if (!@daemons) {
+        @daemons = qw(network nodedev nwfilter secret storage proxy log lock);
+        (is_xen_host) ? push @daemons, 'xen' : push @daemons, 'qemu';
     }
 
     if (is_alp) {
         record_soft_failure("Restarting modular libvirt daemons has not been implemented in ALP. See poo#129086");
     } else {
         # Restart the sockets first
         foreach my $daemon (@daemons) {
-            systemctl("restart $daemon\{,-ro,-admin\}.socket");
+            if (($daemon == 'lock') || ($daemon == 'log')) {
+                systemctl("restart virt${daemon}d\{,-admin\}.socket");
+            } else {
+                systemctl("restart virt${daemon}d\{,-ro,-admin\}.socket");
+            }
         }
 
         # Introduce idle time here (e.g., sleep 5) if necessary
-        sleep 5;
+        script_retry("virsh list", delay => 10, retry => 1, die => 0) for (0 .. 4);
 
         # Restart the services after a brief idle time
         foreach my $daemon (@daemons) {
-            systemctl("restart $daemon.service");
+            systemctl("restart virt${daemon}d.service");
         }
     }
 
-    record_info("Libvirt daemons restarted", join(' ', @daemons));
+    record_info("Modular Libvirt daemons restarted", join(' ', @daemons));
+    check_modular_libvirt_daemons;
```

Comment:
Suggest to add one fatal check here:
```
script_retry("virsh list", delay => 3, retry => 3, die => 1);
```

Reply:
Quick question, what's the purpose to add this fatal check here ?